src/freenet/client/filter/WAVFilter.java

```diff
@@ -0,0 +1,124 @@
+package freenet.client.filter;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+import freenet.l10n.NodeL10n;
+
+public class WAVFilter extends RIFFFilter {
+	// RFC 2361
+	private final int WAVE_FORMAT_UNKNOWN = 0;
+	private final int WAVE_FORMAT_PCM = 1;
+	private final int WAVE_FORMAT_IEEE_FLOAT = 3;
+	private final int WAVE_FORMAT_ALAW = 6;
+	private final int WAVE_FORMAT_MULAW = 7;
+	
+	@Override
+	protected byte[] getChunkMagicNumber() {
+		return new byte[] {'W', 'A', 'V', 'E'};
+	}
+	
+	class WAVFilterContext {

```

Comment:
This class is not used elsewhere, please make it private.

Additionally, the fields in this class should not be declared `public` as they are not effectively accessible outside this class.
Either make them `private`, or of you prefer a more compact look, make them package-private (by just omitting `public`).